testdata/Makefile

```diff
@@ -29,3 +41,42 @@ loader-%-eb.elf: loader.c
 
 %-eb.elf : %.c
 	$(CLANG) $(CFLAGS) -mbig-endian -c $< -o $@
+
+help: ## Display this help.
+	@grep -E '^[a-zA-Z_-]+:.*## .*$$' $(MAKEFILE_LIST) | \
+		awk 'BEGIN {FS = ":.*## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
+
+else
+
+# If the first make goal contains `docker`, only declare Docker-related targets.
+
+.PHONY: docker docker-shell
+
+# Get the absolute path and uid/gid of the repository on disk.
+GITDIR := $(shell git rev-parse --show-toplevel)
+UIDGID := $(shell stat -c '%u:%g' ${GITDIR})
```

Comment:
I think this should just stat `$(CURDIR)` so we can reuse the recipe elsewhere.

Reply:
Could we align on the reusability requirements? Not clear to me wat needs to be done.